src/test/ui/rust-2018/uniform-paths/ambiguity.rs

```diff
@@ -10,10 +10,8 @@
 
 // edition:2018
 
-#![feature(uniform_paths)]
-
 use std::io;
-//~^ ERROR `std` import is ambiguous
+//~^ ERROR `std` is ambiguous
 
 mod std {
     pub struct io;
```

Comment:
Is the `io` member necessary to get the ambiguity error above? Or would just having `mod std` suffice?

Reply:
`io` is not necessary to get the ambiguity (the ambiguities are based on the first segment only).

However, `io` is necessary to not get the additional "no item named `io` in module `std`" error.
Despite the ambiguity being reported later as a non-fatal error, name resolution still considers `use std::...` referring to the `mod std` for all purposes, which is the closest definition in scope (equivalently to non-import paths).